packages/core/lib/commands/test/determineTestFilesToRun.js

```diff
@@ -1,27 +1,65 @@
+const path = require("path");
+const fs = require("fs");
+const glob = require("glob");
+
 const determineTestFilesToRun = ({ inputFile, inputArgs = [], config }) => {
-  const path = require("path");
-  const fs = require("fs");
-  const glob = require("glob");
   let filesToRun = [];
-
   if (inputFile) {
     filesToRun.push(inputFile);
-  } else if (inputArgs.length > 0) {
-    inputArgs.forEach(inputArg => filesToRun.push(inputArg));
+  }else if (inputArgs.length > 0) {
+    //Looping through every argument
+    for(const fileOrDir of inputArgs){
+      let isFileExists = (fs.existsSync(fileOrDir) && fs.statSync(fileOrDir).isFile());
+      let isDirectoryExists = (fs.existsSync(fileOrDir) && fs.statSync(fileOrDir).isDirectory());
+      // If the argument is a file, add the file to filesToRun
+      if(isFileExists){
+        filesToRun.push(fileOrDir);
+      }
+      // if the argument is a directory, run the recursive algorithm to detect every file within the directory and its' subdirectories.
+      else if(isDirectoryExists){
+        const files = getFilesRecursive(fileOrDir,[]);
+        files.forEach((file) => {
```

Comment:
Not necessary but if you want to be slick here do `files.forEach(file => filesToRun.push(file))`. It does return the thing you are adding but it shouldn't matter here.